activerecord/lib/active_record/migration/sql_filesystem_versions.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require "digest"
+require "tempfile"
+require "fileutils"
+
+module ActiveRecord
+  class Migration
+    # Handles schema dumping and loading operations for the :sql_filesystem_versions schema format.
+    # This format keeps the schema in SQL format, but stores migration versions in separate files
+    # to minimize conflicts when merging branches.
+    class SqlFilesystemVersions
```

Comment:
I had an existential crisis naming this, I _badly_ waned to escape `DatabaseTask` - as adding tests there was a bit challenging, file have almost 2k lines already; ended up in `ActiveRecord::Migration` which seems bit _off_